Jira DOC-703: Update Database backup to reflect start time and updated directions. (#1548)
RedisLabs/redislabs-docs: content/rs/administering/import-export/database-backup.md

categories: ["RS"]
aliases: /rs/administering/database-operations/database-backup/
---

> Periodic backups provide a way to restore data with minimal data loss.  With Redis Enterprise Software, you can schedule periodic backups to occur every once a day (every 24 hours), twice a day (every twelve hours), every four hours, or every hour.
> 
> As of v6.2.8, you can specify the start time for twenty four or twelve hour backups.

Reviewer: ```suggestion
Periodic backups provide a way to restore data with minimal data loss.  With Redis Enterprise Software, you can schedule periodic backups to occur once a day (every 24 hours), twice a day (every 12 hours), every four hours, or every hour.

As of v6.2.8, you can specify the start time for 24- or 12-hour backups.
```

Author: Fixed to match intenntion.

Storage location access is verified before periodic backups are scheduled.

To schedule periodic backups for a database:

1.  Sign in to the Redis Enterprise Software admin console using admin credentials.
2.  From the admin console, choose **Databases** and then select your database.
3.  Select the **Edit** button.
4.  Locate and enable the **Periodic backup** checkbox.

    {{<image filename="images/rs/database-configuration-periodic-backup.png" alt="In the admin console, the Periodic backup settings can be found on the Configuration details tab of the database." >}}{{< /image >}}
    
6.  Use the following table to help specify the details:

    | Control name | Description |
    |--------------|-------------|
    | **Interval** | Specifies the frequency of the backup; that is, the time between each backup snapshot.<br/><br/>Supported values include _Every 24 hours_, _Every 12 hours_, _Every 4 hours_, and _Every hour_. |
    | **Set starting time** | _v6.2.8 or later:&nbsp;_ Specifies the start time for the backup; available when **Interval** is set to _Every 24 hours_ or _Every 12 hours_.<br/><br/>If not specified, defaults to a time selected by Redis Enterprise Software. |

>     | **Choose storage type** | Specifies the storage type for the backup.  Supported options vary and might require additional details.  To learn more, see [Supported storage types](#supported-storage-types).

Reviewer: Mismatched anchor
```suggestion
    | **Choose storage type** | Specifies the storage type for the backup.  Supported options vary and might require additional details.  To learn more, see [Supported storage locations](#supported-storage-locations).
```

Author: Fixed

7.  Select **Update** to apply your changes.

Access to the storage location is verified when you apply your updates.  This means the location, credentials, and other details must exist and function before you can enable periodic backups.

## Default backup start time

> If you do _not_ specify a start time for twenty four or twelve hour backups, Redis Enterprise Software chooses one for you, based on the time the backups are enabled.

Reviewer: ```suggestion
If you do _not_ specify a start time for 24- or 12-hour backups, Redis Enterprise Software chooses one for you, based on the time the backups are enabled.
```

Author: Fixed to intention

This choice assumes that your database is deployed to a multi-tenant cluster containing multiple databases.  This means that default start times are staggered (offset) to ensure availability.  This is done by calculating a random offset which specifies a number of seconds added to the start time.  

Here's how it works:

- Assume you're enabling the backup at 4:00 pm (1600 hours).
- You choose to back up your database every 12 hours.
- Because you didn't set a start time, the cluster randomly chooses an offset of 4,320 seconds (or 72 minutes).

> This means your first periodic backup will run roughly around 5:12 pm (72 minutes after the time you enabled backups) and repeats every twelve hours at about that same time.

Reviewer: ```suggestion
This means your first periodic backup will run roughly around 5:12 pm (72 minutes after the time you enabled backups) and repeats every 12 hours at about that same time.
```

Author: Reworked.

Where:

- *protocol*: the server's protocol, can be either `ftp` or `ftps`.
- *username*: your username, if needed.
- *password*: your password, if needed.
- *hostname*: the hostname or IP address of the server.
- *port*: the port number of the server, if needed.
- *path*: the backup path, if needed.

Example: `ftp://username:password@10.1.1.1/home/backups/`

The user account needs permission to write files to the server.

### SFTP server

Before enabling backups to an SFTP server, make sure that:

- Your Redis Enterprise cluster can connect and authenticate to the SFTP server.
- The user specified in the SFTP server location has read and write privileges.
- The SSH private keys are specified correctly.  You can use the key generated by the cluster or specify a custom key.

    When using the cluster auto generated key, copy the **Cluster SSH Public Key** to the appropriate location on the SFTP server.  This is available from the **General** tab of the **Settings** menu in the admin console.

    Use the server documentation to determine the appropriate location for the SSH Public Key.

To backup to an SFTP server, enter the SFTP server location in the format:

```sh

> sftp://user:password@host:<:custom_port>/path/

Reviewer: ```suggestion
sftp://user:password@host:<custom_port>/path/
```

Author: Fixed-ish.  (There should only be one colon).

```

For example: `sftp://username:password@10.1.1.1/home/backups/`

### Local mount point

> Before you choose to backup to a local mount point, make sure that:

Reviewer: ```suggestion
Before you choose to back up to a local mount point, make sure that:
```

Author: Fixed.

- The node has network connectivity to the destination server of the mount point.
- The `redislabs:redislabs` user has read and write privileges on the local mount point
and on the destination server.
- The backup location has enough disk space for your backup files. The backup files
are saved with filenames that include the timestamp so that backup files are not overwritten.

> To backup to a local mount point for a node:

Reviewer: ```suggestion
To back up to a local mount point for a node:
```

Author: Fixed (and cleaned up a bit of nearby language).

    For example: `/mnt/Public`

1. Verify that the user running Redis Enterprise Software has permissions to access and update files in the mount location.

### AWS Simple Storage Service 

To store backups in an Amazon Web Services (AWS) Simple Storage Service (S3) [bucket](https://docs.aws.amazon.com/AmazonS3/latest/userguide/creating-buckets-s3.html):

1.  Sign in to the [AWS Management Console](https://console.aws.amazon.com/).

1.  Use the **Services** menu to locate and select **Storage** > **S3**.  This takes you to the Amazon S3 admin panel.

1.  If you do not already have a bucket for backups, select the **Create Bucket** button in the upper, right corner of the **Buckets** panel.

    1.  When the **Create bucket** screen appears, enter a name for your bucket.

    1.  Set **AWS Region** to an appropriate region.

    1.  Set other properties according to your company standards.

    1.  When finished, select the **Create bucket** button near the bottom of the screen.

1.  Use the Buckets list to locate and select your bucket.  When the settings appear, select the **Permissions** tab, locate the **Access control list (ACL)** section, and then select the **Edit** button.

> 1.  When the **Edit access control list (ACL)** screen appears, locate the Access for other AWS accounts section and then select the **Add grantee** button.

Reviewer: ```suggestion
1.  When the **Edit access control list (ACL)** screen appears, locate the **Access for other AWS accounts** section and then select the **Add grantee** button.
```

Author: Fixed.

>     1.  In the **Grantee** field, enter:

Reviewer: ```suggestion
    1.  In the **Grantee** field, enter the AWS account ID:
```

Author: Fixed.

    
    ```
    fd1b05415aa5ea3a310265ddb13b156c7c76260dbc87e037a8fc290c3c86b614
    ```

    1.  In the **Objects** list, enable **Write**.
    1.  In the **Bucket ACL** list, enable **Read** and **Write**.
    1.  When finished, select the **Save changes** button.

> Once the bucket is available and the permissions are set, use the name of your bucket as the **Backup destination** for your database's Remote backup settings. For example, suppose your bucket is named *backups-bucket*.  In that case, set **Backup destination** to `s3://backups-bucket`.

Reviewer: ```suggestion
Once the bucket is available and the permissions are set, use the name of your bucket as the **Backup destination** for your database's **Remote backup settings**. For example, suppose your bucket is named *backups-bucket*.  In that case, set **Backup destination** to `s3://backups-bucket`.
```

Author: Reworked.

### GCP Storage 

For [Google Cloud Platform (GCP)
console](https://developers.google.com/console/) subscriptions, store your backups in a Google Cloud Storage bucket:

1. Sign in to Google Cloud Platform console.

> 1. In the admin console menu, locate the _Storage_ section than select **Cloud Storage&nbsp;>&nbsp;Browser**.

Reviewer: ```suggestion
1. In the admin console menu, locate the _Storage_ section then select **Cloud Storage&nbsp;>&nbsp;Browser**.
```

Author: Fixed.

1. Verify that your bucket does _not_ have a set retention policy.  

    To do so:

    1. View the details of your bucket.

    1. Select the **Retention** tab.
    
    1. Verify that there is no retention policy.  
    
    If a policy is defined and you cannot delete it, you need to use a different bucket.

Use the bucket details **Configuration** tab to locate the **gsutil URI**.  This is the value you'll assign to your resource's backup path.

### Azure Blob Storage 

To store your backup in Microsoft Azure Blob Storage, sign in to the Azure portal and then:

1. [Create an Azure Storage account](https://docs.microsoft.com/en-us/azure/storage/common/storage-account-create) if you do not already have one

1. [Create a container](https://docs.microsoft.com/en-us/azure/storage/blobs/storage-quickstart-blobs-portal#create-a-container) if you do not already have one

1. [Manage storage account access keys](https://docs.microsoft.com/en-us/azure/storage/common/storage-account-keys-manage)

Set your resource's **Backup Path** to the path of your storage account.

The syntax for creating the backup varies according to your authorization mechanism.  For example:

> `abs://:storage_account_access_key@storage_account_name/container_name/[path/]`

Reviewer: ```suggestion
`abs://storage_account_access_key@storage_account_name/container_name/[path/]`
```

Author: Fixed.
